### DISCUSSION TITLE
ssh into ubuntu from mac backspace cursor moves forward

### DISCUSSION THREAD

**Discussion by gmm1990** [ghostty-org/ghostty#3583] [Q&A] [answered]
I used homebrew to install its build 8469 on sequoia 15.2 . tried ssh into two different ubuntu 20 instances and when I backspace text the cursor moves forward instead of back. It only goes forward the distance of the text entered. I just downloaded ghostly to test out lmk if more information would be helpful

**Comment by mitchellh** [comment, marked as answer]
https://ghostty.org/docs/help/terminfo

**Reply by phillip-haydon** [reply]
Will this bug be fixed?

**Reply by mitchellh** [reply]
It's not a bug, it's just how the terminfo system works unfortunately. Ghostty's terminfo has been accepted into ncurses meaning as distros update their ncurses then they'll automatically know about Ghostty. This can take years (it's up to each distro). That's just how it goes.

There are much more modern approaches to terminfo that don't require this propagation problem (i.e. XTGETTCAP). Ghostty supports this for software that decides to support this. Bash, Zsh, etc. could but don't. Take that up with them!

We're also considering ssh helpers (but you'd have to type like `ghostty ssh`) to transfer our terminfo automatically. 

This is chicken and egg problem that every new terminal has to go through.
